x/auth/ante/validator_tx_fee.go

```diff
@@ -51,8 +51,8 @@ func CheckTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx, simulate bo
 	return feeCoins, priority, nil
 }
 
-func GetMinimumGasPricesWanted(globalMinimumGasPrices, validatorMinimumGasPrices sdk.DecCoins) sdk.DecCoins {
-	return globalMinimumGasPrices.UnionMax(validatorMinimumGasPrices)
+func GetMinimumGasPricesWantedSorted(globalMinimumGasPrices, validatorMinimumGasPrices sdk.DecCoins) sdk.DecCoins {
+	return globalMinimumGasPrices.UnionMax(validatorMinimumGasPrices).Sort()
```

Comment:
I think this was the reason why when @udpatil tested sorted ordered local min gas prices it was still failing, when combined it would not preserve the sorted order. 

Not really needed with us switching to iterative ordering but this is consistent with what Gaia does: https://github.com/cosmos/gaia/blob/main/x/globalfee/ante/fee_utils.go#LL51C1-L51C1